test/controllers/stocks_controller_test.rb

```diff
@@ -0,0 +1,19 @@
+require "test_helper"
+
+class StocksControllerTest < ActionDispatch::IntegrationTest
+  test "should get index" do
+    @stock = create(:stock)
+    @user = create(:user)
+    sign_in @user
+    get stocks_url
+    assert_response :success
+  end
```

Comment:
Does using local variables work here? Also might be good to add spaces between the stages of the test for readability.

```
stock = create(:stock)
user = create(:user)
sign_in user

get stocks_url

assert_response :success
```